[Firebird] Filters doesn't work in Grouping panel

### Description

Filter by value doesn't work in Grouping panel for firebird


### DBeaver Version

23.0.3

### Operating System

win10

### Database and driver

Jaybird JCA/JDBC driver 4.0.0
Firebird 2.5 LI-V2.5.4.26856

### Steps to reproduce


https://user-images.githubusercontent.com/31996417/233621862-8f78f4cf-bd13-42d4-9ef6-236be1a5e8ea.mp4



### Additional context

```
eclipse.buildId=unknown
java.version=17.0.6
java.vendor=Eclipse Adoptium
BootLoader constants: OS=win32, ARCH=x86_64, WS=win32, NL=en
Command-line arguments:  -os win32 -ws win32 -arch x86_64

org.jkiss.dbeaver.model
Error
Fri Apr 21 12:58:36 CEST 2023
Dynamic SQL Error; SQL error code = -104; Invalid command; no column name specified for column number 2 in derived table GBQ_ [SQLState:42000, ISC error code:336397220]

java.sql.SQLSyntaxErrorException: Dynamic SQL Error; SQL error code = -104; Invalid command; no column name specified for column number 2 in derived table GBQ_ [SQLState:42000, ISC error code:336397220]
	at org.firebirdsql.gds.ng.FbExceptionBuilder$Type$1.createSQLException(FbExceptionBuilder.java:534)
	at org.firebirdsql.gds.ng.FbExceptionBuilder.toFlatSQLException(FbExceptionBuilder.java:304)
	at org.firebirdsql.gds.ng.wire.AbstractWireOperations.readStatusVector(AbstractWireOperations.java:140)
	at org.firebirdsql.gds.ng.wire.AbstractWireOperations.processOperation(AbstractWireOperations.java:204)
	at org.firebirdsql.gds.ng.wire.AbstractWireOperations.readSingleResponse(AbstractWireOperations.java:171)
	at org.firebirdsql.gds.ng.wire.AbstractWireOperations.readResponse(AbstractWireOperations.java:155)
	at org.firebirdsql.gds.ng.wire.AbstractWireOperations.readGenericResponse(AbstractWireOperations.java:257)
	at org.firebirdsql.gds.ng.wire.AbstractFbWireDatabase.readGenericResponse(AbstractFbWireDatabase.java:201)
	at org.firebirdsql.gds.ng.wire.version11.V11Statement.prepare(V11Statement.java:89)
	at org.firebirdsql.jdbc.FBStatement.prepareFixedStatement(FBStatement.java:881)
	at org.firebirdsql.jdbc.FBStatement.internalExecute(FBStatement.java:868)
	at org.firebirdsql.jdbc.FBStatement.executeImpl(FBStatement.java:496)
	at org.firebirdsql.jdbc.FBStatement.execute(FBStatement.java:482)
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCStatementImpl.execute(JDBCStatementImpl.java:329)
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCStatementImpl.lambda$0(JDBCStatementImpl.java:131)
	at org.jkiss.dbeaver.utils.SecurityManagerUtils.wrapDriverActions(SecurityManagerUtils.java:96)
	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCStatementImpl.executeStatement(JDBCStatementImpl.java:131)
	at org.jkiss.dbeaver.ui.controls.resultset.panel.grouping.GroupingDataContainer.readData(GroupingDataContainer.java:120)
	at org.jkiss.dbeaver.ui.controls.resultset.ResultSetJobDataRead.lambda$0(ResultSetJobDataRead.java:123)
	at org.jkiss.dbeaver.model.exec.DBExecUtils.tryExecuteRecover(DBExecUtils.java:189)
	at org.jkiss.dbeaver.ui.controls.resultset.ResultSetJobDataRead.run(ResultSetJobDataRead.java:121)
	at org.jkiss.dbeaver.ui.controls.resultset.ResultSetViewer$ResultSetDataPumpJob.run(ResultSetViewer.java:5092)
	at org.jkiss.dbeaver.model.runtime.AbstractJob.run(AbstractJob.java:105)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
Caused by: org.firebirdsql.jdbc.FBSQLExceptionInfo: Dynamic SQL Error


```

--- a/plugins/org.jkiss.dbeaver.ext.firebird/src/org/jkiss/dbeaver/ext/firebird/model/FireBirdSQLDialect.java
+++ b/plugins/org.jkiss.dbeaver.ext.firebird/src/org/jkiss/dbeaver/ext/firebird/model/FireBirdSQLDialect.java
@@ -83,7 +83,7 @@ public class FireBirdSQLDialect extends GenericSQLDialect {
     }
 
     @Override
-    public boolean supportsAliasForHavingSyntax() {
+    public boolean supportsAliasInHaving() {
         return false;
     }
 

--- a/plugins/org.jkiss.dbeaver.ext.firebird/src/org/jkiss/dbeaver/ext/firebird/model/FireBirdSQLDialect.java
+++ b/plugins/org.jkiss.dbeaver.ext.firebird/src/org/jkiss/dbeaver/ext/firebird/model/FireBirdSQLDialect.java
@@ -83,6 +83,11 @@ public class FireBirdSQLDialect extends GenericSQLDialect {
     }
 
     @Override
+    public boolean supportsAliasForHavingSyntax() {
+        return false;
+    }
+
+    @Override
     public boolean validIdentifierPart(char c, boolean quoted) {
         return super.validIdentifierPart(c, quoted) || c == '$';
     }
@@ -95,10 +100,5 @@ public class FireBirdSQLDialect extends GenericSQLDialect {
     @Override
     public boolean supportsInsertAllDefaultValuesStatement() {
         return true;
-    }
-
-    @Override
-    public boolean supportsAliasInConditions() {
-        return false;
     }
 }

--- a/plugins/org.jkiss.dbeaver.model.sql/src/org/jkiss/dbeaver/model/sql/SQLGroupingQueryGenerator.java
+++ b/plugins/org.jkiss.dbeaver.model.sql/src/org/jkiss/dbeaver/model/sql/SQLGroupingQueryGenerator.java
@@ -162,7 +162,7 @@ public class SQLGroupingQueryGenerator {
 
         if (isDefaultGrouping && showDuplicatesOnly) {
             sql.append("\nHAVING ");
-            if (dataSource.getSQLDialect().supportsAliasForHavingSyntax()) {
+            if (dataSource.getSQLDialect().supportsAliasInHaving()) {
                 sql.append(funcAliases[0]);
             } else {
                 // very special case

--- a/plugins/org.jkiss.dbeaver.model.sql/src/org/jkiss/dbeaver/model/sql/SQLGroupingQueryGenerator.java
+++ b/plugins/org.jkiss.dbeaver.model.sql/src/org/jkiss/dbeaver/model/sql/SQLGroupingQueryGenerator.java
@@ -161,7 +161,14 @@ public class SQLGroupingQueryGenerator {
         boolean isDefaultGrouping = groupFunctions.size() == 1 && groupFunctions.get(0).equals(DEFAULT_FUNCTION);
 
         if (isDefaultGrouping && showDuplicatesOnly) {
-            sql.append("\nHAVING ").append(funcAliases[0]).append(" > 1");
+            sql.append("\nHAVING ");
+            if (dataSource.getSQLDialect().supportsAliasForHavingSyntax()) {
+                sql.append(funcAliases[0]);
+            } else {
+                // very special case
+                sql.append(DEFAULT_FUNCTION);
+            }
+            sql.append(" > 1");
         }
         return sql.toString();
     }

--- a/plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/sql/StandardSQLDialectQueryGenerator.java
+++ b/plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/sql/StandardSQLDialectQueryGenerator.java
@@ -157,7 +157,13 @@ public class StandardSQLDialectQueryGenerator implements SQLDialectQueryGenerato
                         );
                     } else {
                         // Most likely it is an expression so we don't want to quote it
-                        attrName = binding.getMetaAttribute().getName();
+                        String metaName = binding.getMetaAttribute().getName();
+                        if (CommonUtils.isNotEmpty(metaName)) {
+                            attrName = binding.getMetaAttribute().getName();
+                        } else {
+                            // Second option for some databases (like Firebird)
+                            attrName = binding.getMetaAttribute().getLabel();
+                        }
                     }
                 }
             } else if (cAttr != null) {

--- a/plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/sql/SQLDialect.java
+++ b/plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/sql/SQLDialect.java
@@ -270,6 +270,10 @@ public interface SQLDialect {
         return true;
     }
 
+    default boolean supportsAliasForHavingSyntax() {
+        return true;
+    }
+
     boolean supportsTableDropCascade();
 
     boolean supportsOrderByIndex();

--- a/plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/sql/SQLDialect.java
+++ b/plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/sql/SQLDialect.java
@@ -270,7 +270,10 @@ public interface SQLDialect {
         return true;
     }
 
-    default boolean supportsAliasForHavingSyntax() {
+    /**
+     * Checks whether dialect supports alias for queries with HAVING syntax.
+     */
+    default boolean supportsAliasInHaving() {
         return true;
     }
 

